src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -6657,11 +6657,18 @@ def create_public_ip_prefix(cmd, client, resource_group_name, public_ip_prefix_n
         prefix.public_ip_address_version = version if version is not None else 'ipv4'
 
     if cmd.supported_api_version(min_api='2020-06-01') and custom_ip_prefix_name:
-        cip_client = network_client_factory(cmd.cli_ctx).custom_ip_prefixes
-        try:
-            prefix.custom_ip_prefix = cip_client.get(resource_group_name, custom_ip_prefix_name)
-        except ResourceNotFoundError:
-            raise ResourceNotFoundError('Custom ip prefix {} doesn\'t exist.'.format(custom_ip_prefix_name))
+        # support cross-subscription
+        if is_valid_resource_id(custom_ip_prefix_name):
+            custom_ip_prefix_id = custom_ip_prefix_name
+        else:
```

Comment:
You may need aux_subscription client for cross tenant https://github.com/Azure/azure-cli/blob/284d6131679cecc26f928c78f1e4c29e2dd66405/src/azure-cli/azure/cli/command_modules/network/custom.py#L4073

Reply:
Confirmed with @brianlehr that we don't need to support cross-tenant now, and the current behavior is the same as Azure Portal.